### ISSUE TITLE
SDCC copy at end of build broken for Windows

### ISSUE THREAD

**Issue by bbbbbr** [Zal0/gbdk-2020#103]
@untoxa @jserv 

Building on a Win10 x64 VM (that previously built ok) - I'm getting an error on SDCC binary copy at the end. Seems related to the changes merged in PR #88 

Everything else builds fine.

If I revert the copy to the previous method it copies with no problem.
```ifeq ($(OS),Windows_NT)
ifeq ($(OS),Windows_NT)
		@cp -r $(SDCCDIR)/bin $(BUILDDIR)
else
```

SDCCDIR is set as :
SET SDCCDIR=C:\gb\sdcc

Here is the console log
```Installing SDCC
cp: cannot stat `C:gbsdcc/bin/as2gbmap.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/makebin.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/packihx.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdar.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdasgb.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdcc.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdcdb.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdcdb.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdcpp.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdldgb.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdnm.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdobjcopy.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sdranlib.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/sz80.exe': No such file or directory
cp: cannot stat `C:gbsdcc/bin/libgcc_s_seh-1.dll': No such file or directory
cp: cannot stat `C:gbsdcc/bin/libgcc_s_sjlj-1.dll': No such file or directory
cp: cannot stat `C:gbsdcc/bin/libstdc++-6.dll': No such file or directory
cp: cannot stat `C:gbsdcc/bin/libwinpthread-1.dll': No such file or directory
cp: cannot stat `C:gbsdcc/bin/readline5.dll': No such file or directory
make: *** [sdcc-install] Error 1```

**Comment by bbbbbr** [comment]
_Update:_ So if unix path slashes are used then it works again with only some errors. While not particularly intuitive, I guess it works. That should probably be made clear in the Readme and any other docs about building.

Requires: `SET SDCCDIR=C:/gb/sdcc/` instead of what might be expected on windows `SET SDCCDIR=C:\gb\sdcc\`

Note,
`as2gbmap.exe` is not correct and should be `as2gbmap.cmd`

```
Installing SDCC
cp: cannot stat `C:/gb/sdcc//bin/as2gbmap.exe': No such file or directory
-> makebin.exe
-> packihx.exe
-> sdar.exe
-> sdasgb.exe
-> sdcc.exe
-> sdcdb.exe
-> sdcdb.exe
-> sdcpp.exe
-> sdldgb.exe
-> sdnm.exe
-> sdobjcopy.exe
-> sdranlib.exe
-> sz80.exe
cp: cannot stat `C:/gb/sdcc//bin/libgcc_s_seh-1.dll': No such file or directory
cp: cannot stat `C:/gb/sdcc//bin/libgcc_s_sjlj-1.dll': No such file or directory
cp: cannot stat `C:/gb/sdcc//bin/libstdc++-6.dll': No such file or directory
cp: cannot stat `C:/gb/sdcc//bin/libwinpthread-1.dll': No such file or directory
-> readline5.dll
```

**Comment by untoxa** [comment]
as2gbmap.cmd is actually a python tool. we already have not working astorgb.pl, do we actually need such a zoo? where we can use maps in those formats? BGB can load symbols, not maps, should we add noi2sym python tool too? or make binary tools instead?

**Comment by jserv** [comment]
> Requires: `SET SDCCDIR=C:/gb/sdcc/` instead of what might be expected on windows `SET SDCCDIR=C:\gb\sdcc\`

Should [cygpath](https://cygwin.com/cygwin-ug-net/cygpath.html) be used for converting paths?

**Comment by bbbbbr** [comment]
> > Requires: `SET SDCCDIR=C:/gb/sdcc/` instead of what might be expected on windows `SET SDCCDIR=C:\gb\sdcc\`
> 
> Should [cygpath](https://cygwin.com/cygwin-ug-net/cygpath.html) be used for converting paths?

Automatic conversion sounds ok to me. I didn't see cygpath in my Mingw folder though, is it standard? Instead of potentially adding a new executable, what about a make built-in such as subst()? (if it's available. seems like it should be). Something like: `FIXEDPATH := $(subst \\,/,$(WINSTYLEPATH)) `


> as2gbmap.cmd is actually a python tool. we already have not working astorgb.pl, do we actually need such a zoo? where we can use maps in those formats? BGB can load symbols, not maps, should we add noi2sym python tool too? or make binary tools instead?

Removing tools that aren't used sounds good. :+1: 
When possible portable binary tools are nicer, but python seems ok in the mean time. 

I'd like to start building a directory / list of useful tools for GBDK, so it's easier for people to discover them. That can be for another issue though.

**Comment by untoxa** [comment]
that kind of translation that is performed with python is fragile and i'm not sure it works ok with both formats of map files, that are produced by sdld at the moment.

**Comment by untoxa** [comment]
https://github.com/Zal0/gbdk-2020/pull/109
